src/Symfony/Component/Serializer/Mapping/ClassDiscriminatorMapping.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Mapping;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+final class ClassDiscriminatorMapping
```

Comment:
we don't make classes "final" by default, eg for data objects. Is it the case here? If there is no special reason, I'd suggest to remove them.

Reply:
Yes, this is a data object. Happy to remove the `final` even tho I believe closed-first is the best approach 😜

Reply:
@nicolas-grekas removed the final 👍